lessons/lesson2.md

```diff
@@ -1,129 +1,38 @@
-## Lesson 2: Write your app
+## Lesson 2: Verify the result
 
-# H1
-## H2
-### H3
-#### H4
-##### H5
-###### H6
+### Deep dive in
+Now let's take a deeper look into these actions:
+- `bound_package` : default param package created as binding of the shared package deployed in adobe namespace and having the validate action inside itself. Now, this bound_package will also have the validate_action residing in it due to binding.
+- `acp` - package created which will keep the new sync_event_handler sequence
+- `sync_Event_handler_7Z5KH5vv6X` - the new event handler unique to this registration with webhook url
+- `3rd_party_custom_events_3C9419175E9D393C0A495E39@AdobeOrg_2a0237a4-f0d3-45e9-8179-10ab21ef929c_eventrt_7Z5KH5vv6X` - the user sequence created taking the validate action as first action and user runtime action as second action. Created with the same unique identifier suffixed at the end of its name to bind this user sequence with the event registration in 1:1 fashion.
 
-Emphasis, aka italics, with *asterisks* or _underscores_.
 
-Strong emphasis, aka bold, with **asterisks** or __underscores__.
+### Verfiy the result
+Now, we will ingest some events to see how this new SYNC webhook behaves, send through `publish-event` or using postman 
+after ingesting the event, this is the request to the sync webhook, I made one successful one and another failed one to show the debug tracking feature
+![debug-13](assets/debug-tracing-13.png)
 
-Combined emphasis with **asterisks and _underscores_**.
+ Now we can see json response returned from the main event handler webhook in the debug tracing
+![debug-14](assets/debug-tracing-14.png)
 
-Strikethrough uses two tildes. ~~Scratch this.~~
 
-1. First ordered list item
-2. Another item
-⋅⋅* Unordered sub-list. 
-1. Actual numbers don't matter, just that it's a number
-⋅⋅1. Ordered sub-list
-4. And another item.
+The debug tracing response in case of user action code failure. Note that, now the json response includes the activation id of the failed user_sequence (using which user can trace it down to his action code failure) along with the exact user code failure message.
```

Comment:
Ok, I understand that the error is produced to demonstrate the usage of "Debug tracing". But the fact that it returns 500 here is superficial. We should explain why it fails here, why there is a 500, 401, 400, and how to properly handle it so that the event is delivered successfully. The explanation [on this page](https://www.adobe.io/apis/experienceplatform/events/docs.html#!adobedocs/adobeio-events/master/webhook/runtime_webhooks.md) seems pretty complete to me, so we should copy it over. wdyt?

Reply:
sure @duynguyen  good point! we could copy this one over to keep it consistant.